Rest/Controller/ContentController.php

```diff
@@ -278,4 +292,53 @@ private function getRelation($content, $language)
 
         return false;
     }
+
+    /**
+     * Returns author of the content.
+     *
+     * @param Content $contentValue
+     * @param ContentType $contentType
+     *
+     * @return string
+     */
+    private function getAuthor(Content $contentValue, ContentType $contentType)
+    {
+        $author = $contentValue->getFieldValue($this->getFieldIdentifier('author', $contentType));
+
+        if (null === $author) {
+            $user = $this->userService->loadUser($contentValue->contentInfo->ownerId);
+            $author = $user->contentInfo->name;
+        }
+
+        return (string) $author;
+    }
+
+    /**
+     * Returns field name.
+     *
+     * To define another field name for specific value (e. g. author) add it to default_settings.yml
+     *
+     * For example:
+     *
+     *     ez_recommendation.field_identifiers:
+     *         author:
+     *             blog_post: authors
+     *         image:
+     *             blog_post: thumbnail
+     *
+     * @param string $fieldName
+     * @param ContentType $contentType
+     *
+     * @return string
+     */
+    private function getFieldIdentifier($fieldName, $contentType)
+    {
+        $contentTypeName = $contentType->identifier;
+        $fieldDefinitions = $this->container->getParameter('ez_recommendation.field_identifiers');
+        if (isset($fieldDefinitions[$fieldName]) && in_array($contentTypeName, array_keys($fieldDefinitions[$fieldName]))) {
```

Comment:
this can be far more efficient by not having to use array functions at all, you can simply:
```if (isset($fieldDefinitions[$fieldName]) && !empty($fieldDefinitions[$fieldName][contentTypeName])) {```